### PR TITLE
[Bugfix][SwipeableFlatList] Fix SwipeableFlatList on close

### PR DESCRIPTION
**Pull request by kesha-antonov**
## Motivation

This commit https://github.com/facebook/react-native/commit/d8cc6e3c2b8c2dc25b6b32691779790c15de8d10 introduced ```SwipeableFlatList```

It has bug - doesn't close on swipe other rows.

I fixed it.

## Release Notes

[IOS/ANDROID] [BUGFIX] [SwipeableFlatList] - opened rows did not close on swipe other rows

### CODE REVIEW

**Comment by eballeste** [comment]
What if we don't want the other rows to close automatically?

**Comment by kesha-antonov** [comment]
@eballeste https://github.com/kesha-antonov/react-native/blob/d50cbaae89435be46f91f8c6b62550714fdc4f98/Libraries/Experimental/SwipeableRow/SwipeableFlatList.js#L54

**Comment by kesha-antonov** [comment]
Read source. It's by design

**Comment by AntonPuko** [comment]
@kesha-antonov can you check, I think with your current solution all rows are rerendered on every swipe.

File: Libraries/Experimental/SwipeableRow/SwipeableFlatList.js [hunk 1]
```diff
@@ -35,7 +35,9 @@ type SwipableListProps = {
 type Props<ItemT> = SwipableListProps & FlatListProps<ItemT>;
 
 type State = {
-  openRowKey: ?string,
+  extraData: {
```
Comment (AntonPuko) [inline Libraries/Experimental/SwipeableRow/SwipeableFlatList.js]: why do we need extra object inside state? isn't it simpler keep it as before but put into flat list  `extraData={this.state}` ?

Comment (tomasreimers) [reply Libraries/Experimental/SwipeableRow/SwipeableFlatList.js]: @kesha-antonov 

Following the example in https://facebook.github.io/react-native/docs/flatlist.html

> By passing extraData={this.state} to FlatList we make sure FlatList itself will re-render when the state.selected changes. Without setting this prop, FlatList would not know it needs to re-render any items because it is also a PureComponent and the prop comparison will not show any changes.

File: Libraries/Experimental/SwipeableRow/SwipeableFlatList.js [hunk 2]
```diff
@@ -88,31 +90,37 @@ class SwipeableFlatList<ItemT> extends React.Component<Props<ItemT>, State> {
...
     this.state = {
-      openRowKey: null,
+      extraData: {
+        openRowKey: null,
+      }
     };
 
     this._shouldBounceFirstRowOnMount = this.props.bounceFirstRowOnMount;
+
+    this.onRefFlatList = this.onRefFlatList.bind(this);
```
Comment (tomasreimers) [inline Libraries/Experimental/SwipeableRow/SwipeableFlatList.js]: Why this change?

Comment (kesha-antonov) [reply Libraries/Experimental/SwipeableRow/SwipeableFlatList.js]: This code works faster than fat arrow.
In render we should avoid create function every time
@tomasreimers

Comment (tomasreimers) [reply Libraries/Experimental/SwipeableRow/SwipeableFlatList.js]: @kesha-antonov totally get that -- I meant why .bind(this) in the initializer instead of using class fields (like _onScroll)

**Comment by vysotsky** [comment]
Waiting for this fix to be merged? Any updates on it?

File: Libraries/Experimental/SwipeableRow/SwipeableFlatList.js (hunk 2 above)
Comment (hramos) [reply Libraries/Experimental/SwipeableRow/SwipeableFlatList.js]: Ping

**Comment by kesha-antonov** [comment]
ping

**Comment by tomasreimers** [comment]
Hi @kesha-antonov , I think this was fixed by https://github.com/facebook/react-native/pull/18001
